sklearn/manifold/t_sne.py

```diff
@@ -672,6 +676,12 @@ def __init__(self, n_components=2, perplexity=30.0,
         self.method = method
         self.angle = angle
 
+        # TODO sklearn 0.20: remove this warning
+        if learning_rate == 200.0:
```

Comment:
As mentioned above I feel that makes things worse because:
- it obscures the actual default, which will be 200
- it confuses the user, because `learning_rate` is a float but gets defaulted to a string
- it worsers the function signature

I think there is nothing wrong in warning everybody for a development cycle, a year from now we won't even be talking about this anymore.